packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -34,30 +35,6 @@ export const MenuButton = compose<MenuButtonType>({
       setShowContextualMenu(!showContextualMenu);
     }, [showContextualMenu, setShowContextualMenu]);
 
-    const menuItemsUpdated = menuItems.map((item) => {
-      if (item.hasSubmenu) {
-        const [showSubmenu, setShowSubmenu] = useState(false);
-
-        const toggleShowSubmenu = React.useCallback(() => {
-          setShowSubmenu(!showSubmenu);
-        }, [showSubmenu, setShowSubmenu]);
-
-        const onDismissSubmenu = React.useCallback(() => {
-          setShowSubmenu(false);
-        }, [setShowSubmenu]);
-        const { onHoverIn = toggleShowSubmenu, submenuProps = {}, ...restItems } = item;
-        const { onDismiss = onDismissSubmenu, setShowMenu = toggleShowSubmenu, ...restSubmenuProps } = submenuProps;
-        const menuItemUpdated = {
-          ...restItems,
-          onHoverIn,
-          showSubmenu: item.showSubmenu ? showSubmenu : undefined,
-          submenuProps: { ...restSubmenuProps, onDismiss, setShowMenu },
-        };
-        return menuItemUpdated;
-      }
-      return item;
-    });
-
```

Comment:
Why did this go away?

Reply:
It's being handled by the Submenu component below on line 127.

The issue is that hooks cannot be called inside callbacks, so I needed to move all the hook code into its own component.

Reply:
This code is being handled by the SubMenu component on line 127. The issue is that hooks cannot be called directly inside of callbacks, so I needed to move the code inside a component in order for this to work.

Reply:
OK. Locally testing this seems fine, so yay!

Reply:
> looks like this logic was merged until the function component SubMenuItem below

zzzz I seriously detest the ```start review``` functionality of leaving everything pending and not updating any comments mid-review.